spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/autoconfig/TraceAutoConfiguration.java

```diff
@@ -121,13 +122,21 @@ SpanNamer sleuthSpanNamer() {
 
 	@Bean
 	@ConditionalOnMissingBean
-	Propagation.Factory sleuthPropagation(SleuthProperties sleuthProperties) {
+	BasePropagationFactoryProvider basePropagationFactoryProvider() {
+		return new DefaultBasePropagationFactoryProvider();
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	Propagation.Factory sleuthPropagation(
```

Comment:
Another thing that we could actually do is to 

- create a bean of `ExtraFieldPropagation.FactoryBuilder` 
- we would pass it to `sleuthPropagation`
- that way we don't need a separate interface

WDYT?

Reply:
That sounds much nicer, I'll do a second pass with that. Want me to squash into one commit in the PR?

Reply:
Nah, that's ok.